#!/usr/bin/env python
PACKAGE = "aero_path_planning"

import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("local_frame",   str_t, 0, "The Local Frame for the robot", "/base_footprint")
gen.add("global_frame",  str_t, 0, "The Global Frame for the robot",   "/world")
gen.add("state_topic",   str_t, 0, "The topic for subscribing to AeroState", "/state")
gen.add("path_topic",    str_t, 0, "The topic for subscribing to CarrotPath", "/path")
gen.add("path_threshold",double_t, 0, "Distance, in meters, to consider the robot to have reached a point on the path",  1.0,   0.0, 5.0)
gen.add("mission_goal_topic", str_t, 0, "The topic for publishing new mission goals to", "/mission_goal")
gen.add("path_goal_topic",    str_t, 0, "The topic for publishing new path goals to",    "/path_goal")

exit(gen.generate(PACKAGE, "aero_path_planning", "MissionPlanner"))
